src/data/import.rs

```diff
@@ -76,10 +76,12 @@ fn process_file_chunk(
 
 fn deserialize_chunk(
     headers: &StringRecord,
+    first_index: &usize,
```

Comment:
primitive types shouldn't really be passed by reference. At least when they are small, like an usize is the same size a pointer, so with a reference you pass the same 8 bytes that then need to be dereferenced to get the actual 8 bytes (in case of 64-bit system).

In general all primitive types like numbers in Rust implement the `Copy` trait / marker, see https://doc.rust-lang.org/std/marker/trait.Copy.html

Means these primitives are copied by default, e.g. when passed into a function, instead of moved. So there shouldn't really be a need to clone them by hand (only in this case you have a reference and that isn't copy).

```suggestion
    first_index: usize,
```

Reply:
Oh, that is really good to know!  
I've changed that.